straxen/plugins/event_patternfit.py

```diff
@@ -453,22 +448,28 @@ def lbinom_pmf_mode(x_min, x_max, target, args, err=1e-7, max_iter=50):
         max_iter -= 1
     return x1
 
-
 @numba.njit
 def lbinom_pmf_inverse(x_min, x_max, target, args, err=1e-7, max_iter=50):
     """Find the where the log Binomial pmf cross target with secant method"""
     x0 = x_min
     x1 = x_max
     dx = abs(x1 - x0)
-
-    while (dx > err) and (max_iter > 0):
-        y0 = lbinom_pmf(x0, *args)
-        y1 = lbinom_pmf(x1, *args)
-        dx, x0, x1 = _numeric_derivative(y0, y1, err, target, x_min, x_max, x0, x1)
-        max_iter -= 1
+    
+    if dx != 0:
+        while (dx > err) and (max_iter > 0):
+            y0 = lbinom_pmf(x0, *args)
+            y1 = lbinom_pmf(x1, *args)
+            dx, x0, x1 = _numeric_derivative(y0, y1, err, target, x_min, x_max, x0, x1)
+            max_iter -= 1           
+        if x0 == x1 == 0 and y0 - target > 0:
+            x1 = np.nan
+        if x0 == x1 == args[0] and y0 - target < 0:
+            x1 = np.nan
+    else:
+        x1 = np.nan
+        
     return x1
 
-
 @numba.njit
 def binom_test(k, n, p):
```

Comment:
Yeah, the precision depends on S1. For low S1 we have high tolerance, but this decrease with the increasing of the area. Can we maybe make it as a warning instead of error ? I will then be very clear in the summary note.

Reply:
Yeah, that sounds good (how to turn test into warning?)